invenio_rdm_records/resources/serializers/datacite/schema.py

```diff
@@ -73,18 +75,26 @@ def get_name_identifiers(self, obj):
         """Get name identifier list."""
         serialized_identifiers = []
         identifiers = obj["person_or_org"].get("identifiers", [])
-
+        scheme_uri = ""
```

Comment:
I would personally move these 3 added block codes next to each other, below the `name_id` initialization, to avoid switching context:

```python
                name_id = {
                    "nameIdentifier": identifier["identifier"],
                    "nameIdentifierScheme": id_scheme,
                }

                scheme_uri = ""
                if scheme == "orcid":
                    scheme_uri = "http://orcid.org/"
                elif scheme == "ror":
                    scheme_uri = "https://ror.org/"

                if scheme_uri:
                    name_id["schemeUri"] = scheme_uri
```